src/libopensc/card-oberthur.c

```diff
@@ -2245,14 +2245,16 @@ auth_read_record(struct sc_card *card, unsigned int nr_rec, unsigned int idx,
 	if (flags & SC_RECORD_BY_REC_NR)
 		apdu.p2 |= 0x04;
 
-	apdu.le = count;
-	apdu.resplen = count;
+	apdu.le = MIN(count, SC_MAX_APDU_BUFFER_SIZE);
+	apdu.resplen = SC_MAX_APDU_BUFFER_SIZE;
```

Comment:
AFAIK this (including the `recvbuf` definition above) should have the length of `SC_MAX_APDU_RESP_SIZE`, instead of `SC_MAX_APDU_BUFFER_SIZE` currently. At least according to the comment next to its definition, this size is for encoding the whole APDU, not just the response buffer:

https://github.com/OpenSC/OpenSC/blob/f041813c3fdf76d80605647f34606827a8178784/src/libopensc/types.h#L34

But I am not sure if it is worth fixing it here now, given that this was not problem for last 20 years. Also not sure if this could break something (likely not).

Reply:
I only want to make sure that the buffer is not exceeded. Other semantic optimizations may be possible, but that needs to be tested, which I am unable to do.